### PR TITLE
OUT-821 | Opening "Filter by assignee" drop-down for second time shows incorrect assignees

### PR DESCRIPTION
**Pull request by rrojan**
### Tasks

- [OUT-821 | Opening "Filter by assignee" drop-down for second time shows incorrect assignees](https://linear.app/copilotplatforms/issue/OUT-821/opening-filter-by-assignee-drop-down-for-second-time-shows-incorrect)

### Changes

- [x] It seems we had some legacy code that required us to reset to initial assignee list state on selector input change. This was causing assignee list to revert to the first array after page load everytime we triggered assignee selector. This has been fixed
- [x]  Remove unused imports & Organize imports via TS lsp

### Testing Criteria

- [x] Video:
[Screencast from 2024-09-24 13-43-24.webm](https://github.com/user-attachments/assets/1f913fc1-6d17-4662-a6f4-6a366ea4ab2b)

### Note:
I set up my prettier config to organize imports as well, so a lot of changes are just unused imports that have been purged, or reordered import statements.

### CODE REVIEW

**Review by aatbip — CHANGES_REQUESTED** [review]
The changes in this PR seems like have created side effects hence bringing [THIS](https://linear.app/copilotplatforms/issue/OUT-769/filter-dropdown-search-input-persists-until-i-click-somewhere-on-the) issue back again. Please consider fixing.

**Comment by rrojan** [comment]
@aatbip I tested this thoroughly and the issue doesn't seem to occur. Could you please check again, or add a loom for this

**Comment by aatbip** [comment]
> @aatbip I tested this thoroughly and the issue doesn't seem to occur. Could you please check again, or add a loom for this

Please see the below LOOM video which demonstrates the issue which was resolved for [THIS](https://linear.app/copilotplatforms/issue/OUT-769/filter-dropdown-search-input-persists-until-i-click-somewhere-on-the) task - 

[LOOM](https://www.loom.com/share/8f47be6f3db4444cb327171650fd441f?sid=564acfd8-336a-40ca-9e0a-980ffc132283)

@rrojan

**Review by aatbip — CHANGES_REQUESTED** [review]
https://github.com/copilot-platforms/tasks-app/pull/311#issuecomment-2378397410

**Comment by rrojan** [comment]
@aatbip Thanks for the loom. Seems like the task you linked to is a different one - so I couldn't repro it (it's about the search text input not getting cleared out)

File: src/components/layouts/FilterBar.tsx [hunk 1]
```diff
@@ -16,6 +16,7 @@ import { CreateViewSettingsDTO } from '@/types/dto/viewSettings.dto'
 import { FilterOptions, FilterOptionsKeywords, IAssigneeCombined, IFilterOptions } from '@/types/interfaces'
 import { filterOptionsToAssigneeMap, filterTypeToButtonIndexMap } from '@/types/objectMaps'
 import { checkAssignee } from '@/utils/assignee'
+import { stopPendingFetch } from '@/utils/fetchSideEffects'
```
Comment (aatbip) [inline src/components/layouts/FilterBar.tsx:19]: I think this is creating another issue. The previous issue was about search result persisting when dropdown is toggled again. 
Now the issue is - dropdown doesn't show any result at all when I open it from the second time onwards. Please kindly check the video below - 

https://www.loom.com/share/fe1c7b01502046ffb628068d63804537?sid=94b29072-cb37-4623-aa5e-16277edf3ec2

**Review by aatbip — CHANGES_REQUESTED** [review]

**Pull request closed without merging**